kedro/framework/cli/cli.py

```diff
@@ -210,7 +210,8 @@ def main(
                 project_metadata=self._metadata, command_args=args, exit_code=1
             )
             hook_called = True
-            raise
+            traceback.print_exc()
+            sys.exit(1)
```

Comment:
Now it looks like we don't need `hook_called` flag anymore as all the branches that set it to true now have `sys.exit(1)`. So maybe we can just move `self._cli_hook_manager.hook.after_command_run` to the `try` block and get rid of `finally`, so the logic is a bit cleaner?

Reply:
I don't want to block the PR, so we can pay attention to this in a separate one.

Reply:
Actually, I think you're right we don't need the `finally` block at all. If the the command executes successfully or the command doesn't exist - it goes into the first `except` block otherwise if there's an exception it is caught by the next `except`. Hook gets called either way.

Reply:
Check if it doesn't get called twice on a successful case too, it's why that `finally` block was introduced.